samples/routing/custom/sample.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Custom routing
+summary: Customizing NServiceBus message routing
+component: Core
+tags:
+- Routing
+redirects:
+---
+
+The sample demonstrates how NServiceBus routing model can be extended to allow for configuration-free routing with MSMQ transport. It does so by making endpoint instances publish metadata information about themselves
+ * what is my identify (logical endpoint name and physical address)
+ * what messages can I handle
+ * what event types do I publish
+
+The advantage of such approach is low development friction (no configuration to keep up to date for environment). The disadvantage is the added complexity and failure modes due to problems with the discovery mechanism. 
```

Comment:
> due to problems with the discovery mechanis

Sounds like those have to occurs ;) Not sure if that's exactly what you were trying to say, but I'm trying ;)

The disadvantage is that the configuration (or endpoint?) discovery mechanism implementation is more complex, especially with regards to failure handling mechanism.